core/src/test/scala/org/apache/spark/sql/delta/ShowTableColumnsSuite.scala

```diff
@@ -0,0 +1,147 @@
+      result: DataFrame,
+      expected: Seq[Seq[Any]],
+      columns: Seq[String]): Unit = {
+    checkAnswer(
+      result.select(columns.head, columns.tail: _*),
+      expected.map { x => Row(x: _*)})
+  }
+
+  private def showDeltaColumnsTest(
+      fileToTableNameMapper: File => String,
+      schemaName: String = ""): Unit = {
+    withDatabase("delta") {
+      val tempDir = Utils.createTempDir()
+      Seq(1 -> 1)
+        .toDF("column1", "column2")
+        .write
+        .format("delta")
+        .mode("overwrite")
+        .save(tempDir.toString)
+
+      val finalSchema = if (schemaName.nonEmpty) s"FROM $schemaName" else ""
+      checkResult(sql(s"SHOW COLUMNS IN ${fileToTableNameMapper(tempDir)} $finalSchema"),
+        outputColumnValues,
+        outputColumnNames)
+    }
+  }
+
+  // when no schema name provided, default schema name is `default`.
+  test("delta table: table identifier") {
+    showDeltaColumnsTest(f => s"delta.`${f.toString}`", schemaName = "")
+  }
+
+  test("delta table: table name with separated schema name") {
+    showDeltaColumnsTest(f => s"`${f.toString}`", schemaName = "delta")
+  }
+
+  test("non-delta table: table identifier with catalog table") {
+    // Non-Delta table represent by catalog identifier (e.g.: sales.line_ite) is supported in
+    // SHOW COLUMNS command.
+    List("json", "csv", "orc", "parquet").foreach { x =>
+      withTable("show_columns") {
+        sql(s"""
+               |CREATE TABLE show_columns(column1 INT, column2 INT)
+               |USING $x
+               |COMMENT "describe a non delta table"
+        """.stripMargin)
+        checkResult(sql("SHOW COLUMNS IN show_columns"), outputColumnValues, outputColumnNames)
+      }
+    }
+  }
+
+  test("delta table: table name not found") {
+    val fakeTableName = s"test_table"
+    val schemaName = s"delta"
+    showDeltaColumnsTest(f => s"$schemaName.`${f.toString}`", schemaName = "")
+    val e = intercept[AnalysisException] { sql(s"SHOW COLUMNS IN `$fakeTableName` IN $schemaName") }
+    assert(e.getMessage().contains(s"Table or view not found: $schemaName.$fakeTableName"))
+  }
+
+  test("delta table: duplicated schema name") {
+    // If there are 2 schema name in the command, the one attached on table name will be ignored.
+    // e.g.: `SHOW COLUMNS delta.test_table IN epsilon` == `SHOW COLUMNS test_table IN epsilon`.
+    val schemaName = s"epsilon"
+    withDatabase(schemaName) {
+      sql(s"CREATE DATABASE $schemaName")
+      val tableName = "test_table"
+      withTable(tableName) {
+        sql(s"""
+             |CREATE TABLE $schemaName.$tableName(column1 INT, column2 INT)
+             |USING delta
+          """.stripMargin)
+        checkResult(
+          sql(s"SHOW COLUMNS IN delta.$tableName IN $schemaName"),
+          outputColumnValues,
+          outputColumnNames)
+      }
+    }
+  }
+
+  testWithTempView(s"delta table: show columns on temp view") { isSQLTempView =>
```

Comment:
View on delta table doesn't use our implementation, right? Could you update the test name to avoid confusions? such as `show columns on temp view should fallback to Spark`. In order to test that, it's better to make the temp view a subset of the delta table so that we can verify we are not calling our implementation.

Reply:
Updated the test case title to `show columns on temp view should fallback to Spark`